auth_jwt/models/ir_http.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+import re
+
+from odoo import SUPERUSER_ID, api, models, registry as registry_get
+from odoo.http import request
+
+from ..exceptions import (
+    AccessDeniedMalformedAuthorization,
+    AccessDeniedMissingAuthorization,
+    AccessDeniedSessionMismatch,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+AUTHORIZATION_RE = re.compile(r"^Bearer ([^ ]+)$")
+
+
+class IrHttpJwt(models.AbstractModel):
+
+    _inherit = "ir.http"
+
+    @classmethod
+    def _auth_method_jwt(cls, validator_name=None):
+        assert request.db
+        assert not request.uid
+        token = cls._get_bearer_token()
+        assert token
+        registry = registry_get(request.db)
+        with registry.cursor() as cr:
+            env = api.Environment(cr, SUPERUSER_ID, {})
+            validator = env["auth.jwt.validator"]._get_validator_by_name(validator_name)
+            assert len(validator) == 1
```

Comment:
Is-it safe to use assert here? Why not a simple exception raise?

Reply:
because _get_validator_by_name raises AccessDenied if anything looks suspicious. This assert is just an assertion that _get_validator_by_name did what it should.